controllers/articles.controller.js

```diff
@@ -38,3 +40,24 @@ exports.getCommentsByArticleId = (req, res, next) => {
         })
         .catch(next);
 };
+
+exports.postCommentById = (req, res, next) => {
+    const { article_id } = req.params;
+    const { user_name, body } = req.body;
+
+    const promises = [insertCommentsByArticleId(article_id, body)];
+
+    if (article_id) {
+        promises.push(checkArticleExist(article_id, "articles", "article_id"));
+    }
+    if (user_name) {
+        promises.push(checkArticleExist(user_name, "users", "username"));
+    }
```

Comment:
author in the comments table is a foreign key which relates to username in the users table. A psql error will be thrown if this is violated. It is not necessary to separately check if the user exists.